extensions/pkg/controller/worker/genericactuator/actuator_reconcile.go

```diff
@@ -182,27 +190,34 @@ func (a *genericActuator) Reconcile(ctx context.Context, worker *extensionsv1alp
 		}
 	}
 
-	if rollingUpdate {
-		deployment := &appsv1.Deployment{}
-		if err := a.client.Get(ctx, kutil.Key(worker.Namespace, v1beta1constants.DeploymentNameClusterAutoscaler), deployment); err != nil {
-			if !apierrors.IsNotFound(err) {
-				return err
-			}
-		} else {
-			if err := util.ScaleDeployment(ctx, a.client, deployment, 1); err != nil {
-				return err
-			}
+	if clusterAutoscalerUsed && !isHibernated {
+		if err = a.scaleClusterAutoscaler(ctx, worker, 1); err != nil {
+			return err
 		}
 	}
 
-	if err := a.updateWorkerStatusMachineDeployments(ctx, worker, wantedMachineDeployments); err != nil {
+	if err := a.updateWorkerStatusMachineDeployments(ctx, worker, wantedMachineDeployments, false); err != nil {
 		return errors.Wrapf(err, "failed to update the machine deployments in worker status")
 	}
 
 	return nil
 }
 
-func (a *genericActuator) deployMachineDeployments(ctx context.Context, cluster *controller.Cluster, worker *extensionsv1alpha1.Worker, existingMachineDeployments *machinev1alpha1.MachineDeploymentList, wantedMachineDeployments worker.MachineDeployments, classKind string, clusterAutoscalerUsed bool) error {
+func (a *genericActuator) scaleClusterAutoscaler(ctx context.Context, worker *extensionsv1alpha1.Worker, replicas int32) error {
```

Comment:
Hm, ideally, the generic worker actuator does not scale the cluster-autoscaler anymore. It shouldn't know anything about it. Instead, it should simply set the `RollingUpdate` condition to `True` and then a controller in g/g can decide whether to scale down the CA. WDYT?
However, I see that it's difficult to do this in a backwards compatible way. So let's keep this code as is, but maybe you can add a comment that this code can be removed in a future version when such just mentioned controller in g/g is implemented.

Reply:
Yes that what I was referring to in the description. I agree with what you are saying, that was the idea.

> Please note that the cluster autoscaler scale down behaviour has not changed (scaled down during creation and rolling updates).
In the future, the cluster autoscaler might not be required to be scaled down any more.
See this issue and the corresponding issue on the MCM

In this PR I just did not want to change the behaviour when the scale down happens. This should be done when the external controller is ready.